src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php

```diff
@@ -34,6 +34,21 @@ public function __construct(EntityManager $entityManager, ClassMetadata $class)
         $this->associationHydrator = new AssociationHydrator($entityManager, $class);
     }
 
+    public function findAggregate(mixed $id): ?ProductInterface
```

Comment:
Naming wise you will need to do similar to 
https://github.com/Sylius/Sylius/blob/0e6ac906c6e29866d485209aa3a3619a00f3a3ab/src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php#L440

For example
```suggestion
    public function findOneForAdmin($id): ?ProductInterface
```
```suggestion
    public function findOneForAdminCrud($id): ?ProductInterface
```

Reply:
Yup, the `findAggregate` doesn't sound as Sylius naming convention. But I would prefer `findPreloaded` than admin-related name, as It could potentially be used outside admin context.